sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1685,13 +1689,36 @@ private void SetNameFieldsIfNull()
                 operationName: nameof(BlobBaseClient.Download),
                 parameterName: nameof(conditions));
 
+            bool? rangeGetContentMD5 = null;
+            bool? rangeGetContentCRC64 = null;
+            string structuredBodyType = null;
+            switch (validationOptions?.ChecksumAlgorithm.ResolveAuto())
+            {
+                case StorageChecksumAlgorithm.MD5:
+                    rangeGetContentMD5 = true;
+                    break;
+                case StorageChecksumAlgorithm.StorageCrc64:
+                    if (pageRange?.Length <= Constants.StructuredMessage.MaxDownloadCrcWithHeader)
```

Comment:
Hmm, do we want to just always use Structured Message for this download? Is splitting the logic worth the efficiency gain of using the header? I guess the logic split is not very impactful in code, but I wonder if customers who are watching requests might be confused? Not really sure here, just thinking aloud.

Reply:
sticking with this for compatibility on certain error states.